### PR TITLE
perf(core): Optimize executions filtering by metadata

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1577

### CODE REVIEW

**Comment by ivov** [comment]
@krynble CI says `MySQL BLOB/TEXT column 'key' used in key specification without a key length`, which indicates that MySQL requires specifying a length for the new index for `execution_metadata.key`, but that column (or the FE for that matter) [does not apply any length limits](https://github.com/n8n-io/n8n/blob/82fe6383ef48f3c6b06dc76ce41a54f81b631a41/packages/cli/src/databases/migrations/mysqldb/1679416281779-CreateExecutionMetadataTable.ts#L9). Do we want to add those limits now? Or skip the new index for now?

**Comment by krynble** [comment]
@ivov this is not great, I don't know why we're using `blob/text` for the `key` attribute. It should be something like a `varchar` because size should be limited.

iirc we do impose limits in the FE, but fixing this now would be a bit more complicated.

We can skip adding the index now, to unblock this, but I would still like us to fix this either when we drop mysql (fixed already) or at some point thinking whether we can change the column type.

File: packages/cli/test/integration/shared/db/executions.ts [hunk 1]
```diff
@@ -34,6 +39,16 @@ export async function createExecution(
 		deletedAt,
 	});
 
+	if (metadata?.length === 1) {
```
Comment (krynble) [inline packages/cli/test/integration/shared/db/executions.ts:42]: When saving a new execution it's possible to store multiple key-value pairs as metadata, but the filtering currently only allows to filter by 1 key-value pair at a time.

I'm not sure why this changes was necessary and I don't think checking for `length === 1` would be a good choice either. 

Why was this code added here?

Comment (krynble) [reply packages/cli/test/integration/shared/db/executions.ts:42]: But this is a call to create executions, that's why I'm confused

Comment (ivov) [reply packages/cli/test/integration/shared/db/executions.ts]: Oh silly mixup, exact same line in two separate files and I thought you meant the other. Please see [`2286578`](https://github.com/n8n-io/n8n/pull/9477/commits/228657806bc758b164f89a007c08bb09c8727934).

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.44.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.44.0)
